File: AppKitExample/src/main/java/me/grishka/appkit/example/fragments/StackTestFragment.java
package me.grishka.appkit.example.fragments;

import android.app.Activity;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import me.grishka.appkit.Nav;
import me.grishka.appkit.example.R;
import me.grishka.appkit.fragments.ToolbarFragment;

public class StackTestFragment extends ToolbarFragment{

	private int index;

	@Override
	public void onAttach(Activity activity){
		super.onAttach(activity);
		setTitle("Fragment stack");
		index=getArguments().getInt("index");
	}

	@Override
	public View onCreateContentView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
		View view=inflater.inflate(R.layout.stack_test_fragment, container, false);

		TextView text=view.findViewById(R.id.text);
		Button btn=view.findViewById(R.id.start_fragment);

		text.setText("Fragment "+index);
		btn.setOnClickListener(new View.OnClickListener(){
			@Override
			public void onClick(View view){
				Bundle args=new Bundle();
				args.putInt("index", index+1);
				Nav.go(getActivity(), StackTestFragment.class, args);
			}
		});

		return view;
	}
}

File: AppKitExample/src/main/java/me/grishka/appkit/example/fragments/StackTestFragment.kt
package me.grishka.appkit.example.fragments

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView

import me.grishka.appkit.Nav
import me.grishka.appkit.example.R
import me.grishka.appkit.fragments.ToolbarFragment

class StackTestFragment : ToolbarFragment() {

    private var index: Int = 0

    override fun onAttach(activity: Activity) {
        super.onAttach(activity)
        setTitle("Fragment stack")
        index = arguments.getInt("index")
    }

    override fun onCreateContentView(inflater: LayoutInflater, container: ViewGroup, savedInstanceState: Bundle): View {
        val view = inflater.inflate(R.layout.stack_test_fragment, container, false)

        val text = view.findViewById<TextView>(R.id.text)
        val btn = view.findViewById<Button>(R.id.start_fragment)

        text.text = "Fragment $index"
        btn.setOnClickListener {
            val args = Bundle()
            args.putInt("index", index + 1)
            Nav.go(activity, StackTestFragment::class.java, args)
        }

        return view
    }
}
